commands/logHw.js

```diff
@@ -73,62 +73,54 @@ module.exports = {
-                    channelID: foundClass.channelID.S,
-                    title: foundClass.title.S,
-                    img: foundClass.image_url.S,
-                    serverID: foundClass.serverID.S
-                };
-
-                console.log('DATA FETCHED');
-
-                // get LogBookChannelID and GuildID of main server
-                messageChannelDB.read(interaction.channel.id)
-                    .then((channel) => {
-                        const messageChannelID = channel.channelID.S;
-                        const messageChannelGuildID = channel.guildID.S;
-                        const guild = interaction.client.guilds.cache.get(messageChannelGuildID);
-                        const messageChannel = guild.channels.cache.get(messageChannelID);
+        const foundClass = await ClassDB.read(classCode);
+        if (!foundClass) {
+            return interaction.followUp(`Class code ${classCode} not found. <a:shookysad:949689086665437184>`);
+        }
+        const classInfo = {
+            assignedRole: foundClass.roleID.S,
+            channelID: foundClass.channelID.S,
+            title: foundClass.title.S,
+            img: foundClass.image_url.S,
+            serverID: foundClass.serverID.S
+        };
+        console.log('DATA FETCHED');
 
-                        // Search in the db for all the homework submitted and checked during a period of time
-                        HomeworkDB.read(classInfo.channelID, startDay, endDay, classCode)
-                            .then((homework) => {
-                                console.log(homework);
+        // get LogBookChannelID and GuildID of main server
+        const channel = await messageChannelDB.read(interaction.channel.id);
+        const messageChannelID = channel.channelID.S;
+        const messageChannelGuildID = channel.guildID.S;
+        const guild = interaction.client.guilds.cache.get(messageChannelGuildID);
+        const messageChannel = guild.channels.cache.get(messageChannelID);
 
-                                const alreadyLoggedStudentIds = [];
-                                homework.sort().reverse().map((hw) => {
-                                    const hwNumber = hw.type.S;
-                                    const studentId = hw.studentID.S;
-                                    const hasStudentAlreadyBeenLogged = alreadyLoggedStudentIds.includes(studentId);
-                                    if (shouldNotAllowMultipleEntries && hasStudentAlreadyBeenLogged) {
-                                        return;
-                                    }
+        // Search in the db for all the homework submitted and checked during a period of time
+        const homeworks = await HomeworkDB.read(classInfo.channelID, startDay, endDay, classCode);
+        console.log(homeworks);
+        const alreadyLoggedStudentIds = [];
+        homeworks.sort((a, b) => a.timestamp - b.timestamp).map((hw) => {
```

Comment:
just wanted to double check -- we want to sort in ascending order by datetime (oldest first) or descending order (most recent first)? Also is the `timestamp` a string or DateTime obj?  Just wanted to make sure that we don't actually need to convert the timestamp into a datetime obj, if it's a string there's a chance that it's just coincidentally sorting properly